src/deriver/surprise_reasoner.py

```
@@ -0,0 +1,511 @@
+            "total_duration_ms": total_duration_ms,
+            "convergence_reason": convergence_reason
+        }
+
+    def _adjust_context_for_depth(self):
+        """
+        Adjust the embedding store's observation retrieval counts based on recursion depth
+        to progressively focus on the most relevant observations.
+        """
+        if not self.embedding_store:
+            return
+            
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)